cmd/tempo/app/modules.go

```diff
@@ -146,8 +146,11 @@ func (t *App) initQuerier() (services.Service, error) {
 		}
 	}
 
+	// do not enable polling if this is the single binary. in that case the compactor will take care of polling
```

Comment:
Slightly confused by this. Enabling polling will enable tenant index creation as well. Should we always allow the compactors to create the tenant index? If the queriers also try to create the index they can overwrite the index created by the compactors?

Reply:
> Would it be a better idea to only allow compactors to create the tenant index?
Currently only compactors create the index no matter the mode. The querier always just pulls the tenant index.

In the single binary if both the compactor and querier enable polling I think it will just duplicate work. There will be 2 goroutines in the single binary: One creating the tenant index and one pulling it. It honestly might not be worth this code to get the querier component to not poll.

Reply:
> Currently only compactors create the index no matter the mode.

Umm, correct me if I'm wrong, but I feel even the queriers are attempting to create the tenant index:
- Queriers will call [pollTenantAndCreateIndex](https://github.com/grafana/tempo/blob/17867a66e5488f63671b75d4d41fd439507ee8c1/tempodb/blocklist/poller.go#L123)
- which in-turn calls [buildTenantIndex](https://github.com/grafana/tempo/blob/17867a66e5488f63671b75d4d41fd439507ee8c1/tempodb/blocklist/poller.go#L241).
- and since the poller config will be shared between all components, [TenantIndexBuilders](https://github.com/grafana/tempo/blob/17867a66e5488f63671b75d4d41fd439507ee8c1/tempodb/blocklist/poller.go#L242) will be > 0
- the queriers will then proceed to build the tenant index

Not sure what I'm missing

Reply:
`buildTenantIndex` will always return false b/c the querier returns false here:

https://github.com/grafana/tempo/pull/834/files#diff-a1fec687cd87c7167aee355f831ebae6beb368066e3bfbf092ef0aaf5e216aa2R261

The compactor has logic to check the ring to determine if it should return true or false.

Reply:
Got it. 👍